interface.js

```diff
@@ -0,0 +1,88 @@
+
```

Comment:
Your interface is doing a lot of logic. You would probably benefit from having one more logic class that handles the game, and then the interface talking to that class, without doing any logic.